### DISCUSSION TITLE
V-Select not allowing for customization of item-text & item-value

### DISCUSSION THREAD

**Discussion by Atomic71** [vuetifyjs/vuetify#15011] [Q&A]
Hello all,

First of all - thanks for the amazing work. Vue 2 and Vuetify combo is my bread and butter.

I've recently initialized a project with the 3.0.0-beta release. So far, it's great. I did notice one thing though. It's no longer possible to set `v-select`'s `item-text` & `item-value` props, even though it says so in the documentation (albeit, in the documentation page for the `v-autocomplete`, there is no `v-select` documentation page, but they are more or less the same). So I was looking around in the code, why even the default "text" was not working anymore for me (I mapped all my items to {text: '...', value: '...'}, but was getting nowhere. After looking around, I noticed in the new codebase, it's looking for the title prop. This solved my problem, but I think it'd be nice if vuetify@3 could provide those two props again, like in version 2.

The code lines I'm referring to are [these](https://github.com/vuetifyjs/vuetify/blob/b53301e00f7daef438b2c08568cf6dc3a591bccd/packages/vuetify/src/components/VSelect/VSelect.tsx#L47-L52).

What my question is - can I somehow join the effort, via helping on this particular issue? From what I see, couple of things need to be done:

1) Update docs, `title` is new default (not `text`)
2) Enable passage of `item-text` & `item-value`

I've never contributed to open source before, so please be gentle on me. Apologies if this is not the place to ask this.

**Comment by somewhat3000** [comment]
Could you share an example of how you got this working? I'm stuck on the same problem and title prop doesn't work for me either. Thank you!

**Reply by Atomic71** [reply]
> Could you share an example of how you got this working? I'm stuck on the same problem and title prop doesn't work for me either. Thank you!

I actually wound up going w/ a different library, because I was keen on using Vue 3 and Vuetify is not quite ready yet. It's much laggier for Vue 3 version. I used PrimeVue, which I can only recommend, very good components, and specifically made for Vue 3.

**Reply by Atomic71** [reply]
> > Could you share an example of how you got this working? I'm stuck on the same problem and title prop doesn't work for me either. Thank you!
> 
> I actually wound up going w/ a different library, because I was keen on using Vue 3 and Vuetify is not quite ready yet. It's much laggier for Vue 3 version. I used PrimeVue, which I can only recommend, very good components, and specifically made for Vue 3.

And actually, if I remember correctly, if you use `title` instead of `text`, it should be working fine.

**Reply by somewhat3000** [reply]
I started off a new project with vuetify 3 and I'm strongly considering moving to another framework aswell. I was under the misconception that basic components like select eg were working already. Unfortunately I only realized when working with this, that it is still quite incomplete. I somehow got lured by the release date in May '22. 
For anyone who'd like the answer to my question: 
```
const variable = computed(() => {
		let persons = [];
		someArray(person => {
			person.title = "a title";
			persons.push(person)
		});
		return persons;
	});
```

**Comment by monkut** [comment]
Have the same issue here, trying to use `v-combobox` and `v-autocomplete`, with an array of objects.
And the display is showing `[object Object]`.

Here's what I'm playing with, but it's not working:

```
        <v-combobox
          v-model="addingPlayerInfo"
          :items="players"
          label="name"
          :title="getPlayerDisplayText(item)"
          :item-title="name"
          @change="playerSelectChanged()"
          required
          return-object
        >
        </v-combobox>
```

**Reply by somewhat3000** [reply]
@monkut  What version of the beta are you using? You can check this with `npm list --depth=1`. I am currently using beta6 and do not encounter such problems. I suspect the problem might be on your end in this case. If you could provide a little more code, maybe someone can help (eg the player object and the component in which you use the combobox)

**Comment by tkakar** [comment]
I was wondering if we can group items in v-select. For Vuetify v2, we could have a header item in the items list to get the grouping. Is there an alternative to achieve that with Vuetify 3?
Thanks!

**Comment by c3dsutton** [comment]
I was having this issue and the following works for me 
                <v-select style="width: 20em"
                          v-model="coverageConfigurationDto!.system" 
                          :items="coverageConfigurationDto!.systems"
                          :item-title="item => item.listValue"        
                          return-object
                          label="Select a System"
                          />
